src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+	public static final String CONTEXT_TYPE_USER = 'User';
+	
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is a Sys Admin based on ProfileId.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's Profile.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isSysAdmin(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsSysAdmin = new Map<Id, Boolean>();
+		
+		// Retrieve the SysAdmin Profile Id
+		Id sysAdminId = retrieveSysAdminProfileId();
+
+		// Determine if the User is a SysAdmin
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(sysAdminId == currentUser.ProfileId) {
+					userIdToIsSysAdmin.put(currentUser.Id, true);
+				} else {
+					userIdToIsSysAdmin.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsSysAdmin;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is active based on IsActive field.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsActive status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isActiveUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsActive = new Map<Id, Boolean>();
+		
+		// Determine if the User isActive
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(currentUser.IsActive) {
+					userIdToIsActive.put(currentUser.Id, true);
+				} else {
+					userIdToIsActive.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsActive;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is frozen based on IsFrozen field.
+    * @param updatedUsers The user(s) that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsFrozen status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isFrozenUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsFrozen = new Map<Id, Boolean>();
+
+		// Retrieve the UserLogin records for the updatedUsers
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			Map<Id, Boolean> userLoginUserIdToIsFrozen = retrieveUserLogins(updatedUsers);
+			
+			// Populate the Map with UserId and IsFrozen
+			for(User currentUser : updatedUsers) {
```

Comment:
since retrieveUserLogins() gives you the actual result map you want, what is the reason for then creating a separate map and copying values to it.  seems inefficient and unnecessary.

Reply:
Updated this method based on changes to the retrieveUserLogin method.